### DISCUSSION TITLE
Converting/exporting Eclipse plugin projects into maven artefacts

### DISCUSSION THREAD

**Discussion by Jad-el-khoury** [eclipse-tycho/tycho#3527] [Q&A]
I have a number of Eclipse plugins that  I would also need to use within a new Maven project.

Is there a way I can systematically convert these plugins into Maven artifacts? That is, the logic of these plugins should be available for inclusing in both Maven and plugin projects.

I am already using Tycho to build these plugins with maven. But - to my understanding - this is not the same as producing maven artefacts.

Can Tycho help with that? If not, any other tips?

The closest I could find is [eclipse:make-artifacts](https://maven.apache.org/plugins-archives/maven-eclipse-plugin-2.9/make-artifacts-mojo.html) from 2011 (deprecated)

**Comment by laeubi** [comment]
@Jad-el-khoury if you build these with Tycho they are already maven artifacts so no need to convert those at all and you can simply deploy them to any maven artifact repository you want, did you encounter any issues?

**Reply by Jad-el-khoury** [reply]
Hmmmm...! I did not realize this. and I can indeed include the built artifacts in another maven project.
I was under the impression that Tycho builds using maven, but the end results is still Plugins and not maven artifacts.
 
Thanks @laeubi!

**Reply by laeubi** [reply]
Plugins are jars, Maven artifacts are jars (if packaging = jar) sp it does not really exclude one for the other.
